component/vdr/orb/config/service.go

```diff
@@ -337,10 +338,61 @@ func (cs *Service) getEndpointIPNS(didURI string) (*models.Endpoint, error) {
 		return nil, fmt.Errorf("anchor origin is not ipns")
 	}
 
-	anchorOriginSplit := strings.Split(anchorOrigin, "ipns://")
+	currentAnchorOrigin := anchorOrigin
+
+	for {
+		latestAnchorOrigin, err := cs.getLatestCID(currentAnchorOrigin, didSplit[4])
+		if err != nil {
+			return nil, err
+		}
+
+		if latestAnchorOrigin == currentAnchorOrigin {
+			break
+		}
+
+		currentAnchorOrigin = latestAnchorOrigin
+	}
+
+	anchorOriginSplit := strings.Split(currentAnchorOrigin, "ipns://")
 
 	return cs.populateResolutionEndpoint(fmt.Sprintf("%s/%s/%s", ipfsGlobal, "ipns", anchorOriginSplit[1]),
-		url.PathEscape(anchorOrigin))
+		url.PathEscape(currentAnchorOrigin))
+}
+
+func (cs *Service) getLatestCID(anchorOrigin, suffix string) (string, error) {
+	var webFingerResponse restapi.WebFingerResponse
+
+	anchorOriginSplit := strings.Split(anchorOrigin, "ipns://")
```

Comment:
This also needs to work for https origins.

Reply:
@troyronda done